notebooks/contrib-dev/McCabe_Thiele_Steven_Final.ipynb

```diff
@@ -0,0 +1,1665 @@
+            "50                  0.000000                       0.00\n"
+          ]
+        }
+      ],
+      "source": [
+        "# imports the csv data into python\n",
+        "url = \"https://raw.githubusercontent.com/chennieXD/McCabe-Thiele/main/LiquidVaporEquil.csv\"\n",
+        "liq_vap_data = pd.read_csv(url)\n",
+        "print(liq_vap_data)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "WyG_uHyiYLQr"
+      },
+      "source": [
+        "## 3.2. Vapor-liquid equilibrium of methanol\n",
+        "\n",
+        "Plot the VLE of methanol using the model generated from linear regression using the format $a+bx+cx^2+dx^3+ex^4+fx^5+gx^6$ and define the equation in the function \"VLE_Eq\"  Plot the 45 degree line on the same plot.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "We will do a linear regression fitting of VLE data to a 6th degree polynomial:\n",
+        "\n",
+        "\\begin{equation}\n",
+        "VLE(x) = a + bx + cx^2 + dx^3 + ex^4 + fx^5 + gx^6\n",
+        "\\end{equation}\n",
+        "\n",
+        "where:\n",
+        "\\begin{align*}\n",
+        "x & : \\text{Mole Fraction of Methanol} \\in [0, 1] \\\\\n",
+        "VLE(x) & : \\text{Mole Fraction of Methanol in Vapor} \\\\\n",
+        "a, b, c, d, e, f, g & : \\text{Coefficients to be determined}\n",
+        "\\end{align*}\n",
+        "\n",
+        "The goal is to obtain the coefficients $a, b, c, d, e, f, g$ that best fit the VLE data. The model is fitted using linear regression in matrix form, following these equations:\n",
+        "\n",
+        "\\begin{equation}\n",
+        "\\mathbf{X}^T = \\text{np.transpose}(X)\n",
+        "\\end{equation}\n",
+        "\n",
+        "\\begin{equation}\n",
+        "\\mathbf{XX}^{-1} = \\text{np.linalg.inv}(\\mathbf{X}^T \\cdot X)\n",
+        "\\end{equation}\n",
+        "\n",
+        "\\begin{equation}\n",
+        "\\mathbf{X}^T \\mathbf{Y} = \\mathbf{X}^T \\cdot Y\n",
+        "\\end{equation}\n",
+        "\n",
+        "\\begin{equation}\n",
+        "\\text{coefficients} = \\mathbf{XX}^{-1} \\cdot (\\mathbf{X}^T \\cdot Y)\n",
+        "\\end{equation}\n",
+        "\n",
+        "\n"
+      ],
+      "metadata": {
+        "id": "l5jgZIR5MW5s"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Function for matrix operations\n",
+        "\n",
+        "def calculate_regression_coefficients(X, Y):\n",
+        "    \"\"\"solving for the coefficients of the 6th degree polynomials\n",
+        "\n",
+        "    Arguments:\n",
+        "        X = the polynomial we are trying to fit, based on the liquid vapor fraction\n",
+        "        Y = feature of the data, in this case the vapor fraction\n",
+        "\n",
+        "    Returns:\n",
+        "        Coefficient of each term of the polynomials\n",
+        "\n",
+        "    \"\"\"\n",
```

Comment:
If you want the students/readers to see code, remove the ### BEGIN SOLUTION ### and ### END SOLUTION ### here. This will help us keep the math notation and code separate.